lighthouse-core/lib/network-recorder.js

```diff
@@ -339,8 +339,8 @@ class NetworkRecorder extends EventEmitter {
     // playback all the devtools messages to recreate network records
     devtoolsLog.forEach(message => networkRecorder.dispatch(message));
 
-    // get out the list of records
-    const records = networkRecorder.getRecords();
+    // get out the list of records & filter out invalid records
+    const records = networkRecorder.getRecords().filter(record => record.isValid);
```

Comment:
you might want to add `if (!request || !request.isValid)` to line 323 above too, behavior of the other `on*` methods is unexpected if all the properties are unset

Reply:
Good call.  Fixed!